packages/@react-spectrum/numberfield/src/NumberField.tsx

```diff
@@ -24,7 +24,7 @@ import {useNumberField} from '@react-aria/numberfield';
 import {useNumberFieldState} from '@react-stately/numberfield';
 import {useProviderProps} from '@react-spectrum/provider';
 
-export const NumberField = React.forwardRef((props: SpectrumNumberFieldProps, ref: RefObject<HTMLDivElement>) => {
+export const NumberField = React.forwardRef((props: SpectrumNumberFieldProps & {formatOptions?: Intl.NumberFormatOptions}, ref: RefObject<HTMLDivElement>) => {
```

Comment:
> can we add this to SpectrumNumberFieldProps instead of tagging it in here? or was there a reason for this?

Yeah, sorry the code wasn't finished yet. a lot of things were for testing purposes. Will fix them one by one now )